tests/plugins/event_building.py

```diff
@@ -0,0 +1,35 @@
+from _core import PluginTestAccumulator, PluginTestCase
+import numpy as np
+
+
+@PluginTestAccumulator.register('test_exclude_s1_as_triggering_peaks_config')
+def exclude_s1_as_triggering_peaks_config(self: PluginTestCase):
+    """
+    Test for checking the event building options, specifically the S1 (exclusion)
+    """
+    # Change the config to allow s1s to be triggering, also drastically
+    # decrease the trigger_min_area to allow small s1s to be triggering
+    st = self.st.new_context()
+    st.set_config(dict(exclude_s1_as_triggering_peaks=0,
+                       trigger_min_area=20,
+                       ))
+    events = st.get_array(self.run_id, 'event_basics')
+    new_min_coincidence = int(np.max(events['s1_tight_coincidence']) - 1)
+
+    # Create an alternative config, but with less
+    st_alt = st.new_context()
+    st_alt.set_config(dict(event_s1_min_coincidence=new_min_coincidence))
+    events_alt = st_alt.get_array(self.run_id, 'event_basics')
+
+    # There should be more "alt" events since we don't concatenate as
+    # many peaks into one event
```

Comment:
I think this statement is a bit fragile. You could also have less events because you can trigger on below cathode events etc. I think it would be more robust to really check the triggering conditions like the s1 coincidence and peak area to be larger than the specified settings.

Reply:
Right, a bit of the issue is that it's not super clear which peaks are triggering at the event-basics stage. Any suggestions to this effect? Could probably find some at least based on the extensions the last and first triggering peaks.

Reply:
I think you can ask if the event has either an "s2 larger peak area and smaller n_competing" or an "S1 larger coincidence, peak area and smaller n_competing.

Reply:
but you can still have those peaks - they only won't be triggering. We actually designed the S1 to be incorporated even when only the S2 is triggering - so I think I don't see your point. 

I'll just use the fact that the event is extended left and right to see if a peak is triggering.

Reply:
okay, I just implemented it in [63a0b07](https://github.com/XENONnT/straxen/pull/1022/commits/63a0b0765d2330cb9ff1643dedec5f7761007c2a), I think it's good enough like this

Reply:
> but you can still have those peaks - they only won't be triggering. We actually designed the S1 to be incorporated even when only the S2 is triggering - so I think I don't see your point.

Yes what I meant is that you are basically checking the trigger condition itself. All events must have either a triggering S1 **or** (not and) a triggering S2 with the given conditions.